src/Microsoft.OpenApi.OData.Reader/Common/EdmModelHelper.cs

```diff
@@ -309,24 +310,36 @@ internal static string GenerateODataTypeCastPathOperationIdPrefix(ODataPath path
             string operationId = null;
             if (secondLastSegment is ODataComplexPropertySegment complexSegment)
             {
-                string listOrGet = complexSegment.Property.Type.IsCollection() ? "List" : "Get";
+                string listOrGet = includeListOrGetPrefix ? (complexSegment.Property.Type.IsCollection() ? "List" : "Get") : null;
                 operationId = GenerateComplexPropertyPathOperationId(path, listOrGet);
             }
             else if (secondLastSegment as ODataNavigationPropertySegment is not null || isIndexedCollValuedNavProp)
             {
-                string prefix = "Get";
-                if (!isIndexedCollValuedNavProp &&
+                string listOrGet = null;
+                if (includeListOrGetPrefix)
+                {
+                    if (!isIndexedCollValuedNavProp &&
                     (secondLastSegment as ODataNavigationPropertySegment)?.NavigationProperty.TargetMultiplicity() == EdmMultiplicity.Many)
+                    {
+                        listOrGet = "List";
+                    }
+                    else
+                    {
+                        listOrGet = "Get";
+                    }
+                }
+                else
```

Comment:
This else block can be removed since `listOrGet` is initialized as null in [line 318](https://github.com/microsoft/OpenAPI.NET.OData/pull/386/files#diff-37355bb39b3e21b6b8b386eaa5d7fb0c659e26105b2cff8e9103da0e8a38103eR318).

Reply:
Resolved